src/services/validation.js

```diff
@@ -66,28 +90,51 @@ async function checkUserIdInFirestore(
  * @param {String} dataSource - contains information about where and how the data is stored.
  * @param {String} authType - the authentication method the user has used
  * @param {Number} validateCount - indicates how many times the user has been validated
+ * @param {Object} birthdate - contains the month, day and year
+ * @param {Boolean} isExtraInputValidationRequired - indicates if there are any extra fields being validated
  */
 export async function validateID(
   userID,
   dataSource,
   authType,
-  validateCount = 0
+  validateCount = 0,
+  birthdate,
+  isExtraInputValidationRequired
 ) {
   if (dataSource["type"] == "Firestore") {
-    if (authType == "ID") {
-      return checkUserIdInFirestore(
-        userID,
-        validateCount,
-        dataSource["name"],
-        dataSource["column"]
-      );
-    }
-    if (authType == "OTP") {
-      return checkPhoneNumberInFirestore(
-        userID,
-        dataSource["name"],
-        dataSource["column"]
+    if (isExtraInputValidationRequired) {
+      return (
+        checkBirthdateInFirestore(
```

Comment:
why not a single function to check birthdate+user_id combination? either the pair should exist together -- or we do not allow logging-in.